# HG changeset patch
# User David Meyer <pdox@google.com>
# Date 1306451117 25200
# Branch pnacl-sfi
# Node ID e25c7690d014b1ea5c29383f946a7e219e30f79d
# Parent 7acb57bb414f7f77c2a38791bc521bc203905db1
Revert removing __nacl_read_tp

 From llvm-pnacl-0037-261-e25c7690d014b1ea5c29383f946a7e219e30f79d.patch

 From .hg/patches/llvm-pnacl-0037-261-e25c7690d014b1ea5c29383f946a7e219e30f79d.patch.stripped

diff -r 7acb57bb414f lib/CodeGen/IntrinsicLowering.cpp
--- a/lib/CodeGen/IntrinsicLowering.cpp	Thu May 26 13:02:07 2011 -0700
+++ b/lib/CodeGen/IntrinsicLowering.cpp	Thu Jun 09 16:08:31 2011 -0700
@@ -129,11 +129,8 @@
   // assert that __nacl_read_tp exists.
   // Insert a correctly-typed definition now.
   {
-    // @TODO(pdox): Re-enable checking for __nacl_read_tp when we
-    //              build glibc ourselves.
-
-    //const Type *RetTy = Type::getInt8PtrTy(M.getContext());
-    //M.getOrInsertFunction("__nacl_read_tp", RetTy, (Type*)0);
+    const Type *RetTy = Type::getInt8PtrTy(M.getContext());
+    M.getOrInsertFunction("__nacl_read_tp", RetTy, (Type*)0);
   }
   // @LOCALMOD-END
 
